src/server/input_service.rs

```diff
@@ -441,13 +441,76 @@ enum KeysDown {
     EnigoKey(u64),
 }
 
+/// State for scroll debouncing to filter micro-reversals from trackpad
+/// See: https://github.com/rustdesk/rustdesk/issues/13403
+#[derive(Default)]
+struct ScrollState {
+    last_direction_x: i32, // -1, 0, or 1
+    last_direction_y: i32,
+    last_scroll_time: Option<Instant>,
+}
+
+impl ScrollState {
+    const DEBOUNCE_MS: u64 = 40;
+
+    /// Check if a scroll event should be filtered (debounced)
+    /// Returns true if the event should be dropped
+    fn should_filter(&mut self, x: i32, y: i32) -> bool {
+        let now = Instant::now();
+        let dir_x = x.signum();
+        let dir_y = y.signum();
+
+        let should_drop = if let Some(last_time) = self.last_scroll_time {
+            let elapsed = now.duration_since(last_time).as_millis() as u64;
+
+            // Only filter single-unit opposite-direction scrolls within debounce window
+            if elapsed < Self::DEBOUNCE_MS {
+                let is_micro_reversal_x = dir_x != 0
+                    && self.last_direction_x != 0
+                    && dir_x == -self.last_direction_x
+                    && x.abs() == 1;
+                let is_micro_reversal_y = dir_y != 0
+                    && self.last_direction_y != 0
+                    && dir_y == -self.last_direction_y
+                    && y.abs() == 1;
+
+                is_micro_reversal_x || is_micro_reversal_y
```

Comment:
Maybe it should be `&&`.

The entire event is discarded if either the `X` or `Y` direction is slightly reversed.

**Scenario**: Last scroll was `(0, 5)`, current scroll is `(3, -1)` → `Y` is slightly reversed → the entire event is discarded, but the `X` direction of `3` is a valid scroll.